src/ring_theory/ideal/basic.lean

```diff
@@ -590,6 +590,18 @@ not_is_field_iff_exists_ideal_bot_lt_and_lt_top.trans
     ⟨p, bot_lt_iff_ne_bot.mp (lt_of_lt_of_le bot_lt le_p), hp.is_prime⟩,
    λ ⟨p, ne_bot, prime⟩, ⟨p, bot_lt_iff_ne_bot.mpr ne_bot, lt_top_iff_ne_top.mpr prime.1⟩⟩
 
+lemma _root_.ring.is_field_iff_forall_ideal_eq [nontrivial R] :
```

Comment:
You may replace `comm_ring` by `comm_semiring` on line 554 above (seems to be necessary)?